### PR TITLE
Add method 'hasAccess' to AdminInterface. For next major release.

### PR DESCRIPTION
**Pull request by tim96**
Add method 'hasAccess' to AdminInterface.

### CODE REVIEW

**Comment by tim96** [comment]
@OskarStark Can I ask you a question about milestone? Why milestone is 3.0? 3.0 is current branch @Soullivaneuh write it here #3731.

**Comment by greg0ire** [comment]
I think it is not normal, I fixed that.

File: Admin/Admin.php [hunk 1]
```diff
@@ -2964,6 +2964,27 @@ public function checkAccess($action, $object = null)
     /**
      * {@inheritdoc}
      */
+    public function hasAccess($action, $object = null)
```
Comment (OskarStark) [inline Admin/Admin.php]: this method was already merged, or am i wrong?

Comment (greg0ire) [inline Admin/Admin.php]: You're right, @tim96 should rebase on the tip of master, and it will disappear from the diff.

Comment (tim96) [inline Admin/Admin.php]: You are right. I will rebase this branch and fixed it.

Comment (OskarStark) [inline Admin/Admin.php]: 👍

Comment (tim96) [inline Admin/Admin.php]: Fixed

**Comment by Soullivaneuh** [comment]
Please fill your PR of missing parts indicated by the "Needs" labels.

Feel free to ask if you want some precision of what is needed.

**Comment by tim96** [comment]
@Soullivaneuh hi. can you explain about 'Upgrade note'?

**Comment by greg0ire** [comment]
> hi. can you explain about 'Upgrade note'?

Having a BC-break means that users are supposed to change their code when they upgrade to a new major version. The upgrade note should explain how they are supposed to change their code.

**Comment by tim96** [comment]
@Soullivaneuh @SonataCI @greg0ire Do you have roadmap or date for the next major release? And I will rebase this PR the week before this date.

**Comment by greg0ire** [comment]
> @Soullivaneuh @SonataCI @greg0ire Do you have roadmap or date for the next major release? And I will rebase this PR the week before this date. 

Why? We will not wait until this date to merge your PR, even it contains a BC-break. This is why we have two branches.

As per the roadmap, it is not clearly defined, I think we will release if at least one of this condition is fulfilled : 

1. the previous major release is 6 month old
2. there are a great number (to be defined) of commits in master that are not in stable

But I think in the case of 4.0, the release could come earlier than in 6 months, don't recall the reason though.

**Comment by tim96** [comment]
@greg0ire We have two branches?

> Why?

Please, don't think that I am lazy, but I rebased this PR two times already. Tomorrow you again add something and I need to rebase it again.

**Comment by Soullivaneuh** [comment]
@tim96 Note we just need the rebase, the upgrade note and the changelog note as requested before and the PR will be merged ASAP, even if the next major is not for now.

For the changelog, you can add it on the PR body, following this template: https://github.com/sonata-project/SonataAdminBundle/blob/3.x/.github/PULL_REQUEST_TEMPLATE.md

**Comment by greg0ire** [comment]
I know it's hard but the more you wait the more conflicts you will, since others are working. Not touching files everybody changes should help though, so following @Soullivaneuh 's advice about the CHANGELOG is the way to go. Sorry if it's cumbersome.

**Comment by tim96** [comment]
@Soullivaneuh @greg0ire I rebased this PR. I added some information to changelog and to upgrade file. But I don't find any information how to write upgrade note, so could you please review and give your advice? thank you

File: UPGRADE-3.x.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 UPGRADE 3.x
 ===========
 
+UPGRADE FROM 3.0 to 3.1
+=======================
+
+## AdminInterface
+
+If you're using a custom implementation of `AdminInterface`, you need to add realisation of `hasAccess` method.
```
Comment (greg0ire) [inline UPGRADE-3.x.md]: This is the UPGRADE note and it is good :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -62,4 +63,4 @@ to set the CSS class of dropdown item.
 - Text from Admin's toString method is escaped for html output before adding in flash message to prevent possible XSS vulnerability.
 
 ### Removed
-- Remove `btn-outline` from doctrine-orm-admin form actions buttons
+- Remove `btn-outline` from doctrine-orm-admin form actions buttons
```
Comment (greg0ire) [inline CHANGELOG.md]: Please configure your editor to add a newline at end of file  (will make the red pictogram disappear)

File: UPGRADE-3.x.md (hunk 2 above)
Comment (Soullivaneuh) [inline UPGRADE-3.x.md]: Except one thing. It should be on `UPGRADE-4.0.md`, the next major.

Comment (tim96) [inline UPGRADE-3.x.md]: @greg0ire @Soullivaneuh Thank you. Fixed.

File: CHANGELOG.md (hunk 3 above)
Comment (tim96) [inline CHANGELOG.md]: @greg0ire Thank you. Fixed.

File: UPGRADE-4.0.md [hunk 4]
```diff
@@ -1,6 +1,10 @@
 UPGRADE FROM 3.x to 4.0
 =======================
 
+## AdminInterface
+
+If you're using a custom implementation of `AdminInterface`, you need to add realisation of `hasAccess` method.
```
Comment (Soullivaneuh) [inline UPGRADE-4.0.md]: We already have a note about added method on interface here: https://github.com/sonata-project/SonataAdminBundle/pull/3755/files#diff-b2c918cba71bea146d3145ca75754b85R18

Could you complete this part instead? Thanks.

**Comment by tim96** [comment]
In `changelog` format description on their site I saw they using `from` tag. Can we add it too?

File: UPGRADE-4.0.md (hunk 4 above)
Comment (tim96) [inline UPGRADE-4.0.md]: @Soullivaneuh Thank you for review. Fixed.

**Comment by Soullivaneuh** [comment]
Thank you @tim96! :+1:

**Pull request merged**